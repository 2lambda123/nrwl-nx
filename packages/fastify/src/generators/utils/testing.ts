import { Tree } from '@nrwl/devkit';
import { createTreeWithEmptyWorkspace } from '@nrwl/devkit/testing';

export function createTreeWithFastifyApplication(appName: string): Tree {
  const tree = createTreeWithEmptyWorkspace();
  tree.write(
    'workspace.json',
    String.raw`
      {
        "projects": {
          "${appName}": {
            "root": "apps/api",
            "sourceRoot": "apps/api/src",
            "projectType": "application",
            "prefix": "api",
            "targets":{}
          }
        }
      }
    `
  );

  return tree;
}
