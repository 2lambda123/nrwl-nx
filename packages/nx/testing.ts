export {cold, hot} from 'jasmine-marbles';
export {readAll} from './src/testing-utils';
