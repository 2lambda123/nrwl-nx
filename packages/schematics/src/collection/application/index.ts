import {apply, branchAndMerge, chain, mergeWith, Rule, template, url, SchematicContext} from '@angular-devkit/schematics';
import {Schema} from './schema';
import {strings} from '@angular-devkit/core';
import {RepositoryInitializerTask, NodePackageInstallTask} from '@angular-devkit/schematics/tasks';
import {libVersions} from '../utility/lib-versions';
import {wrapIntoFormat} from '../utility/tasks';

export default function(options: Schema): Rule {
  return wrapIntoFormat((context: SchematicContext) => {
    addSkipInstallAndSkipGit(options, context);

    const npmScope = options.npmScope ? options.npmScope : options.name;
    const templateSource = apply(url('./files'), [
      template({
        utils: strings,
        dot: '.',
        ...libVersions,
        ...(options as object),
        npmScope
      })
    ]);
    return chain([branchAndMerge(chain([mergeWith(templateSource)]))]);
  });
}

function addSkipInstallAndSkipGit(options: Schema, context: SchematicContext) {
  let packageTask;
  if (!options.skipInstall) {
    packageTask = context.addTask(new NodePackageInstallTask(options.directory));
  }
  if (!options.skipGit) {
    context.addTask(new RepositoryInitializerTask(options.directory, options.commit), packageTask ? [packageTask] : []);
  }
}
