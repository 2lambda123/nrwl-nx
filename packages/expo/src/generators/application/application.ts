import {
  formatFiles,
  GeneratorCallback,
  joinPathFragments,
  runTasksInSerial,
  Tree,
} from '@nx/devkit';
import { initGenerator as jsInitGenerator } from '@nx/js';

import { runSymlink } from '../../utils/symlink-task';
import { addLinting } from '../../utils/add-linting';
import { addJest } from '../../utils/add-jest';

import { normalizeOptions } from './lib/normalize-options';
import initGenerator from '../init/init';
import { addProject } from './lib/add-project';
import { createApplicationFiles } from './lib/create-application-files';
import { addEasScripts } from './lib/add-eas-scripts';
import { addDetox } from './lib/add-detox';
import { Schema } from './schema';
import { ensureDependencies } from '../../utils/ensure-dependencies';
import { initRootBabelConfig } from '../../utils/init-root-babel-config';
import { logShowProjectCommand } from '@nx/devkit/src/utils/log-show-project-command';

export async function expoApplicationGenerator(
  host: Tree,
  schema: Schema
): Promise<GeneratorCallback> {
  return await expoApplicationGeneratorInternal(host, {
    addPlugin: false,
    projectNameAndRootFormat: 'derived',
    ...schema,
  });
}

export async function expoApplicationGeneratorInternal(
  host: Tree,
  schema: Schema
): Promise<GeneratorCallback> {
  const options = await normalizeOptions(host, schema);

  const tasks: GeneratorCallback[] = [];
  const jsInitTask = await jsInitGenerator(host, {
    ...schema,
    skipFormat: true,
  });
  tasks.push(jsInitTask);
  const initTask = await initGenerator(host, { ...options, skipFormat: true });
  tasks.push(initTask);
  if (!options.skipPackageJson) {
    tasks.push(ensureDependencies(host));
  }
  initRootBabelConfig(host);

  createApplicationFiles(host, options);
  addProject(host, options);

  const lintTask = await addLinting(host, {
    ...options,
    projectRoot: options.appProjectRoot,
    tsConfigPaths: [
      joinPathFragments(options.appProjectRoot, 'tsconfig.app.json'),
    ],
  });
  tasks.push(lintTask);

  const jestTask = await addJest(
    host,
    options.unitTestRunner,
    options.projectName,
    options.appProjectRoot,
    options.js,
    options.skipPackageJson,
    options.addPlugin
  );
  tasks.push(jestTask);
  const detoxTask = await addDetox(host, options);
  tasks.push(detoxTask);
  const symlinkTask = runSymlink(host.root, options.appProjectRoot);
  tasks.push(symlinkTask);
  addEasScripts(host);

  if (!options.skipFormat) {
    await formatFiles(host);
  }

  tasks.push(() => {
    logShowProjectCommand(options.projectName);
  });

  return runTasksInSerial(...tasks);
}

export default expoApplicationGenerator;
