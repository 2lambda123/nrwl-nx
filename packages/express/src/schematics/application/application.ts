import {
  chain,
  externalSchematic,
  Rule,
  SchematicContext,
  Tree,
  noop,
} from '@angular-devkit/schematics';
import { join, normalize, Path } from '@angular-devkit/core';
import { updateJsonInTree, toFileName, formatFiles } from '@nrwl/workspace';
import { toJS } from '@nrwl/workspace/src/utils/rules/to-js';
import { appsDir } from '@nrwl/workspace/src/utils/ast-utils';
import init from '../init/init';
import { Schema } from './schema';
import { mainModule } from 'process';

interface NormalizedSchema extends Schema {
  appProjectRoot: Path;
}

function addTypes(options: NormalizedSchema): Rule {
  const tsConfigPath = join(options.appProjectRoot, 'tsconfig.app.json');
  return updateJsonInTree(tsConfigPath, (json) => {
    json.compilerOptions.types = [...json.compilerOptions.types, 'express'];
    return json;
  });
}

function addMainFile(options: NormalizedSchema): Rule {
  return (host: Tree) => {
    host.overwrite(
      join(options.appProjectRoot, options.js ? 'src/main.js' : 'src/main.ts'),
      `/**
 * This is not a production server yet!
 * This is only a minimal backend to get started.
 */

import * as express from 'express';

const app = express();

app.get('/api', (req, res) => {
  res.send({ message: 'Welcome to ${options.name}!' });
});

const port = process.env.port || 3333;
const server = app.listen(port, () => {
  console.log(\`Listening at http://localhost:\${port}/api\`);
});
server.on('error', console.error);
`
    );
  };
}

export default function (schema: Schema): Rule {
  return (host: Tree, context: SchematicContext) => {
    const options = normalizeOptions(host, schema);
    return chain([
      init({ ...options, skipFormat: true }),
      externalSchematic('@nrwl/node', 'application', schema),
      addMainFile(options),
      addTypes(options),
      options.js ? toJS() : noop(),
      formatFiles(options),
    ])(host, context);
  };
}

function normalizeOptions(host: Tree, options: Schema): NormalizedSchema {
  const appDirectory = options.directory
    ? `${toFileName(options.directory)}/${toFileName(options.name)}`
    : toFileName(options.name);
  const appProjectRoot = join(normalize(appsDir(host)), appDirectory);

  return {
    ...options,
    appProjectRoot,
  };
}
