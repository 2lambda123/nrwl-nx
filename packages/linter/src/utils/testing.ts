import { SchematicTestRunner } from '@angular-devkit/schematics/testing';
import { join } from 'path';
import { Tree } from '@angular-devkit/schematics';

const migrationTestRunner = new SchematicTestRunner(
  '@nrwl/linter-migrations',
  join(__dirname, '../../migrations.json')
);

const testRunner = new SchematicTestRunner(
  '@nrwl/linter',
  join(__dirname, '../../collection.json')
);

export function runMigration<T = any>(
  migrationName: string,
  options: T,
  tree: Tree
) {
  return migrationTestRunner
    .runSchematicAsync<T>(migrationName, options, tree)
    .toPromise();
}

export function runSchematic<SchemaOptions = any>(
  schematicName: string,
  options: SchemaOptions,
  tree: Tree
) {
  return testRunner.runSchematicAsync(schematicName, options, tree).toPromise();
}
