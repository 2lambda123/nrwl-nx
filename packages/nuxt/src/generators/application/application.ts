import {
  addProjectConfiguration,
  formatFiles,
  generateFiles,
  GeneratorCallback,
  joinPathFragments,
  offsetFromRoot,
  runTasksInSerial,
  toJS,
  Tree,
} from '@nx/devkit';
import { Schema } from './schema';
import nuxtInitGenerator from '../init/init';
import { normalizeOptions } from './lib/normalize-options';
import { createTsConfig } from '../../utils/create-ts-config';
import { getRelativePathToRootTsConfig } from '@nx/js';
import { updateGitIgnore } from '../../utils/update-gitignore';
import { addBuildTarget, addServeTarget } from './lib/add-targets';
import { Linter } from '@nx/eslint';
import { addE2e } from './lib/add-e2e';
import { addLinting } from '../../utils/add-linting';
import { addVitest } from './lib/add-vitest';

export async function applicationGenerator(tree: Tree, schema: Schema) {
  const tasks: GeneratorCallback[] = [];

  const options = await normalizeOptions(tree, schema);

  const outputPath = joinPathFragments('dist', options.appProjectRoot);

  const projectOffsetFromRoot = offsetFromRoot(options.appProjectRoot);

  const nuxtInitTask = await nuxtInitGenerator(tree, {
    ...options,
    skipFormat: true,
  });
  tasks.push(nuxtInitTask);

  addProjectConfiguration(tree, options.name, {
    root: options.appProjectRoot,
    projectType: 'application',
    sourceRoot: `${options.appProjectRoot}/src`,
    targets: {},
  });

  generateFiles(
    tree,
    joinPathFragments(__dirname, './files'),
    options.appProjectRoot,
    {
      ...options,
      offsetFromRoot: projectOffsetFromRoot,
      title: options.projectName,
      dot: '.',
      tmpl: '',
      style: options.style,
    }
  );

  if (options.style === 'none') {
    tree.delete(
      joinPathFragments(options.appProjectRoot, `src/assets/css/styles.none`)
    );
  }

  createTsConfig(
    tree,
    {
      projectRoot: options.appProjectRoot,
      rootProject: options.rootProject,
      unitTestRunner: options.unitTestRunner,
      outputPath,
    },
    getRelativePathToRootTsConfig(tree, options.appProjectRoot)
  );

  addServeTarget(tree, options.name);
  addBuildTarget(tree, options.name, outputPath);

  updateGitIgnore(tree);

  tasks.push(
    await addLinting(tree, {
      projectName: options.projectName,
      projectRoot: options.appProjectRoot,
      linter: options.linter ?? Linter.EsLint,
      unitTestRunner: options.unitTestRunner,
      rootProject: options.rootProject,
    })
  );

  if (options.unitTestRunner === 'vitest') {
    addVitest(tree, options, options.appProjectRoot, projectOffsetFromRoot);
  }

  tasks.push(await addE2e(tree, options));

  if (options.js) toJS(tree);

  if (!options.skipFormat) await formatFiles(tree);

  return runTasksInSerial(...tasks);
}

export default applicationGenerator;
