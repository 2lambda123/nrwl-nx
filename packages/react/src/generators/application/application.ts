import {
  extendReactEslintJson,
  extraEslintDependencies,
} from '../../utils/lint';
import { NormalizedSchema, Schema } from './schema';
import { createApplicationFiles } from './lib/create-application-files';
import { updateSpecConfig } from './lib/update-jest-config';
import { normalizeOptions } from './lib/normalize-options';
import { addProject } from './lib/add-project';
import { addCypress } from './lib/add-cypress';
import { addJest } from './lib/add-jest';
import { addRouting } from './lib/add-routing';
import { setDefaults } from './lib/set-defaults';
import { addStyledModuleDependencies } from '../../rules/add-styled-dependencies';
import {
  addDependenciesToPackageJson,
  convertNxGenerator,
  formatFiles,
  GeneratorCallback,
  joinPathFragments,
  Tree,
  updateJson,
} from '@nrwl/devkit';
import { runTasksInSerial } from '@nrwl/workspace/src/utilities/run-tasks-in-serial';
import reactInitGenerator from '../init/init';
import { Linter, lintProjectGenerator } from '@nrwl/linter';
import { swcCoreVersion } from '@nrwl/js/src/utils/versions';
import { swcLoaderVersion } from '@nrwl/webpack/src/utils/versions';
import { viteConfigurationGenerator, vitestGenerator } from '@nrwl/vite';
import { mapLintPattern } from '@nrwl/linter/src/generators/lint-project/lint-project';

async function addLinting(host: Tree, options: NormalizedSchema) {
  const tasks: GeneratorCallback[] = [];
  if (options.linter === Linter.EsLint) {
    const lintTask = await lintProjectGenerator(host, {
      linter: options.linter,
      project: options.projectName,
      tsConfigPaths: [
        joinPathFragments(options.appProjectRoot, 'tsconfig.app.json'),
      ],
      unitTestRunner: options.unitTestRunner,
      eslintFilePatterns: [
        mapLintPattern(
          options.appProjectRoot,
          '{ts,tsx,js,jsx}',
          options.rootProject
        ),
      ],
      skipFormat: true,
      rootProject: options.rootProject,
    });
    tasks.push(lintTask);

    updateJson(
      host,
      joinPathFragments(options.appProjectRoot, '.eslintrc.json'),
      extendReactEslintJson
    );

    const installTask = await addDependenciesToPackageJson(
      host,
      extraEslintDependencies.dependencies,
      {
        ...extraEslintDependencies.devDependencies,
        ...(options.compiler === 'swc'
          ? { '@swc/core': swcCoreVersion, 'swc-loader': swcLoaderVersion }
          : {}),
      }
    );
    tasks.push(installTask);
  }
  return runTasksInSerial(...tasks);
}

export async function applicationGenerator(host: Tree, schema: Schema) {
  const tasks = [];

  const options = normalizeOptions(host, schema);

  const initTask = await reactInitGenerator(host, {
    ...options,
    skipFormat: true,
    skipBabelConfig: options.bundler === 'vite',
    skipHelperLibs: options.bundler === 'vite',
  });

  tasks.push(initTask);

  createApplicationFiles(host, options);
  addProject(host, options);

  if (options.bundler === 'vite') {
    // We recommend users use `import.meta.env.MODE` and other variables in their code to differentiate between production and development.
    // See: https://vitejs.dev/guide/env-and-mode.html
    host.delete(joinPathFragments(options.appProjectRoot, 'src/environments'));

    const viteTask = await viteConfigurationGenerator(host, {
      uiFramework: 'react',
      project: options.projectName,
      newProject: true,
      includeVitest: true,
    });
    tasks.push(viteTask);
  }

  if (options.bundler !== 'vite' && options.unitTestRunner === 'vitest') {
    const vitestTask = await vitestGenerator(host, {
      uiFramework: 'react',
      project: options.projectName,
      inSourceTests: options.inSourceTests,
    });
    tasks.push(vitestTask);
  }

  if (
    (options.bundler === 'vite' || options.unitTestRunner === 'vitest') &&
    options.inSourceTests
  ) {
    host.delete(
      joinPathFragments(
        options.appProjectRoot,
        `src/app/${options.fileName}.spec.tsx`
      )
    );
  }

  const lintTask = await addLinting(host, options);
  tasks.push(lintTask);

  const cypressTask = await addCypress(host, options);
  tasks.push(cypressTask);

  if (options.unitTestRunner === 'jest') {
    const jestTask = await addJest(host, options);
    tasks.push(jestTask);
  }

  // Handle tsconfig.spec.json for jest or vitest
  updateSpecConfig(host, options);
  const styledTask = addStyledModuleDependencies(host, options.styledModule);
  tasks.push(styledTask);
  const routingTask = addRouting(host, options);
  tasks.push(routingTask);
  setDefaults(host, options);

  if (!options.skipFormat) {
    await formatFiles(host);
  }

  return runTasksInSerial(...tasks);
}

export default applicationGenerator;
export const applicationSchematic = convertNxGenerator(applicationGenerator);
