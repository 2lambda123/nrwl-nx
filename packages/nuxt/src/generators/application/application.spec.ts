import { createTreeWithEmptyWorkspace } from '@nx/devkit/testing';
import { Tree, readJson, readProjectConfiguration } from '@nx/devkit';
import { applicationGenerator } from './application';

describe('app', () => {
  let tree: Tree;
  const name = 'my-app';

  describe('generated files content - as-provided', () => {
    describe('general application', () => {
      beforeAll(async () => {
        tree = createTreeWithEmptyWorkspace();
        await applicationGenerator(tree, {
          name,
          projectNameAndRootFormat: 'as-provided',
          unitTestRunner: 'vitest',
        });
      });
      it('should create all new files in the correct location', async () => {
        const newFiles = tree.listChanges().map((change) => change.path);
        expect(newFiles).toMatchSnapshot();
      });

      it('should add nuxt entries in .gitignore', () => {
        expect(tree.read('.gitignore', 'utf-8')).toMatchSnapshot();
      });

      it('should configure nuxt correctly', () => {
        expect(tree.read('my-app/nuxt.config.ts', 'utf-8')).toMatchSnapshot();
      });

      it('should configure eslint correctly', () => {
        expect(tree.read('my-app/.eslintrc.json', 'utf-8')).toMatchSnapshot();
      });

      it('should configure vitest correctly', () => {
        expect(tree.read('my-app/vitest.config.ts', 'utf-8')).toMatchSnapshot();
        expect(
          tree.read('my-app/tsconfig.spec.json', 'utf-8')
        ).toMatchSnapshot();
        expect(tree.read('my-app/tsconfig.json', 'utf-8')).toMatchSnapshot();
        const packageJson = readJson(tree, 'package.json');
        expect(packageJson.devDependencies['vitest']).toEqual('~0.34.6');
      });

      it('should configure tsconfig and project.json correctly', () => {
        expect(tree.read('my-app/project.json', 'utf-8')).toMatchSnapshot();
        expect(tree.read('my-app/tsconfig.json', 'utf-8')).toMatchSnapshot();
      });
    });

    describe('styles setup', () => {
      beforeAll(async () => {
        tree = createTreeWithEmptyWorkspace();
      });
      it('should configure css', async () => {
        await applicationGenerator(tree, {
          name: 'myapp1',
          projectNameAndRootFormat: 'as-provided',
          unitTestRunner: 'none',
          style: 'css',
        });
        expect(tree.exists('myapp1/src/assets/css/styles.css')).toBeTruthy();
        expect(tree.read('myapp1/nuxt.config.ts', 'utf-8')).toMatchSnapshot();
      });

      it('should configure scss', async () => {
        await applicationGenerator(tree, {
          name: 'myapp2',
          projectNameAndRootFormat: 'as-provided',
          unitTestRunner: 'none',
          style: 'scss',
        });
        expect(tree.exists('myapp2/src/assets/css/styles.scss')).toBeTruthy();
        expect(tree.read('myapp2/nuxt.config.ts', 'utf-8')).toMatchSnapshot();
      });

      it('should configure less', async () => {
        await applicationGenerator(tree, {
          name: 'myapp3',
          projectNameAndRootFormat: 'as-provided',
          unitTestRunner: 'none',
          style: 'less',
        });
        expect(tree.exists('myapp3/src/assets/css/styles.less')).toBeTruthy();
        expect(tree.read('myapp3/nuxt.config.ts', 'utf-8')).toMatchSnapshot();
      });

      it('should not configure styles', async () => {
        await applicationGenerator(tree, {
          name: 'myapp4',
          projectNameAndRootFormat: 'as-provided',
          unitTestRunner: 'none',
          style: 'none',
        });
        expect(tree.exists('myapp4/src/assets/css/styles.css')).toBeFalsy();
        expect(tree.read('myapp4/nuxt.config.ts', 'utf-8')).toMatchSnapshot();
      });
    });

    describe('pcv3', () => {
      let originalValue: string | undefined;
      beforeEach(() => {
        tree = createTreeWithEmptyWorkspace();
        originalValue = process.env['NX_PCV3'];
        process.env['NX_PCV3'] = 'true';
      });

      afterEach(() => {
        if (originalValue) {
          process.env['NX_PCV3'] = originalValue;
        } else {
          delete process.env['NX_PCV3'];
        }
      });

      it('should not add targets', async () => {
        await applicationGenerator(tree, {
          name,
          projectNameAndRootFormat: 'as-provided',
          unitTestRunner: 'vitest',
        });

        const projectConfi = readProjectConfiguration(tree, name);
        expect(projectConfi.targets.build).toBeUndefined();
        expect(projectConfi.targets.serve).toBeUndefined();
        // TODO(katerina): Enable once `@nx/vite/plugin` is released
        // expect(projectConfi.targets.test).toBeUndefined();
      });
    });
  });
});
