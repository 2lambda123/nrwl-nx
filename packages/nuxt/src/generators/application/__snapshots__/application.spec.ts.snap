// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app generated files content - as-provided general application should add nuxt entries in .gitignore 1`] = `
"
# Nuxt dev/build outputs
.output
.data
.nuxt
.nitro
.cache"
`;

exports[`app generated files content - as-provided general application should configure eslint correctly 1`] = `
"{
  "extends": ["@nuxt/eslint-config", "../.eslintrc.json"],
  "ignorePatterns": ["!**/*"],
  "overrides": [
    {
      "files": ["*.ts", "*.tsx", "*.js", "*.jsx", "*.vue"],
      "rules": {
        "vue/multi-word-component-names": "off"
      }
    }
  ]
}
"
`;

exports[`app generated files content - as-provided general application should configure nuxt correctly 1`] = `
"import { nxViteTsPaths } from '@nx/vite/plugins/nx-tsconfig-paths.plugin';
import { defineNuxtConfig } from 'nuxt/config';

// https://nuxt.com/docs/api/configuration/nuxt-config
export default defineNuxtConfig({
  srcDir: 'src',
  buildDir: 'dist/my-app/.nuxt',
  devtools: { enabled: true },
  devServer: {
    host: 'localhost',
    port: 4200,
  },
  typescript: {
    typeCheck: true,
    tsConfig: {
      extends: 'my-app/tsconfig.app.json',
    },
  },
  imports: {
    autoImport: false,
  },

  vite: {
    plugins: [nxViteTsPaths()],
  },
  nitro: {
    output: {
      dir: 'dist/my-app/.output',
      serverDir: 'dist/my-app/.output/server',
      publicDir: 'dist/my-app/.output/public',
    },
  },
});
"
`;

exports[`app generated files content - as-provided general application should configure tsconfig and project.json correctly 1`] = `
"{
  "name": "my-app",
  "$schema": "../node_modules/nx/schemas/project-schema.json",
  "projectType": "application",
  "sourceRoot": "my-app/src",
  "targets": {
    "serve": {
      "executor": "@nx/nuxt:serve",
      "outputs": ["{options.outputPath}"],
      "options": {}
    },
    "build": {
      "executor": "@nx/nuxt:build",
      "outputs": ["{options.outputPath}"],
      "options": {
        "outputPath": "dist/my-app"
      }
    },
    "lint": {
      "executor": "@nx/eslint:lint",
      "outputs": ["{options.outputFile}"],
      "options": {
        "lintFilePatterns": ["my-app/**/*.{ts,tsx,js,jsx,vue}"]
      }
    },
    "test": {
      "executor": "@nx/vite:test",
      "outputs": ["{options.reportsDirectory}"],
      "options": {
        "passWithNoTests": true,
        "reportsDirectory": "../coverage/my-app",
        "config": "my-app/vitest.config.ts"
      }
    }
  }
}
"
`;

exports[`app generated files content - as-provided general application should configure tsconfig and project.json correctly 2`] = `
"{
  "compilerOptions": {},
  "files": [],
  "include": [],
  "references": [
    {
      "path": "./tsconfig.app.json"
    },
    {
      "path": "./tsconfig.spec.json"
    }
  ],
  "extends": "../tsconfig.base.json"
}
"
`;

exports[`app generated files content - as-provided general application should configure vitest correctly 1`] = `
"import { defineVitestConfig } from 'nuxt-vitest/config';
import { nxViteTsPaths } from '@nx/vite/plugins/nx-tsconfig-paths.plugin';

export default defineVitestConfig({
  plugins: [nxViteTsPaths()],
  test: {
    globals: true,
    cache: {
      dir: '../node_modules/.vitest',
    },
    include: ['my-app/src/**/*.{test,spec}.{js,mjs,cjs,ts,mts,cts,jsx,tsx}'],
    environment: 'nuxt',
  },
});
"
`;

exports[`app generated files content - as-provided general application should configure vitest correctly 2`] = `
"{
  "extends": "./tsconfig.json",
  "compilerOptions": {
    "outDir": "../dist/out-tsc",
    "types": ["vitest/globals", "vitest/importMeta", "vite/client", "node"],
    "composite": true
  },
  "include": [
    "vitest.config.ts",
    "src/**/*.test.ts",
    "src/**/*.spec.ts",
    "src/**/*.test.tsx",
    "src/**/*.spec.tsx",
    "src/**/*.test.js",
    "src/**/*.spec.js",
    "src/**/*.test.jsx",
    "src/**/*.spec.jsx",
    "src/**/*.d.ts"
  ]
}
"
`;

exports[`app generated files content - as-provided general application should configure vitest correctly 3`] = `
"{
  "compilerOptions": {},
  "files": [],
  "include": [],
  "references": [
    {
      "path": "./tsconfig.app.json"
    },
    {
      "path": "./tsconfig.spec.json"
    }
  ],
  "extends": "../tsconfig.base.json"
}
"
`;

exports[`app generated files content - as-provided general application should create all new files in the correct location 1`] = `
[
  ".prettierrc",
  "package.json",
  "nx.json",
  "tsconfig.base.json",
  ".prettierignore",
  "my-app/project.json",
  "my-app/.npmrc",
  "my-app/nuxt.config.ts",
  "my-app/src/app.vue",
  "my-app/src/components/NxWelcome.vue",
  "my-app/src/pages/about.vue",
  "my-app/src/pages/index.vue",
  "my-app/src/public/.gitkeep",
  "my-app/src/public/favicon.ico",
  "my-app/src/server/api/greet.ts",
  "my-app/src/server/tsconfig.json",
  "my-app/tsconfig.app.json",
  "my-app/tsconfig.json",
  ".gitignore",
  ".eslintrc.json",
  ".eslintignore",
  "my-app/.eslintrc.json",
  "my-app/vitest.config.ts",
  "my-app/tsconfig.spec.json",
  "my-app-e2e/project.json",
  "my-app-e2e/src/e2e/app.cy.ts",
  "my-app-e2e/src/support/app.po.ts",
  "my-app-e2e/src/support/e2e.ts",
  "my-app-e2e/src/fixtures/example.json",
  "my-app-e2e/src/support/commands.ts",
  "my-app-e2e/cypress.config.ts",
  "my-app-e2e/tsconfig.json",
  "my-app-e2e/.eslintrc.json",
]
`;

exports[`app generated files content - as-provided styles setup should configure css 1`] = `
"import { nxViteTsPaths } from '@nx/vite/plugins/nx-tsconfig-paths.plugin';
import { defineNuxtConfig } from 'nuxt/config';

// https://nuxt.com/docs/api/configuration/nuxt-config
export default defineNuxtConfig({
  srcDir: 'src',
  buildDir: 'dist/myapp1/.nuxt',
  devtools: { enabled: true },
  devServer: {
    host: 'localhost',
    port: 4200,
  },
  typescript: {
    typeCheck: true,
    tsConfig: {
      extends: 'myapp1/tsconfig.app.json',
    },
  },
  imports: {
    autoImport: false,
  },

  css: ['~/assets/css/styles.css'],

  vite: {
    plugins: [nxViteTsPaths()],
  },
  nitro: {
    output: {
      dir: 'dist/myapp1/.output',
      serverDir: 'dist/myapp1/.output/server',
      publicDir: 'dist/myapp1/.output/public',
    },
  },
});
"
`;

exports[`app generated files content - as-provided styles setup should configure less 1`] = `
"import { nxViteTsPaths } from '@nx/vite/plugins/nx-tsconfig-paths.plugin';
import { defineNuxtConfig } from 'nuxt/config';

// https://nuxt.com/docs/api/configuration/nuxt-config
export default defineNuxtConfig({
  srcDir: 'src',
  buildDir: 'dist/myapp3/.nuxt',
  devtools: { enabled: true },
  devServer: {
    host: 'localhost',
    port: 4200,
  },
  typescript: {
    typeCheck: true,
    tsConfig: {
      extends: 'myapp3/tsconfig.app.json',
    },
  },
  imports: {
    autoImport: false,
  },

  css: ['~/assets/css/styles.less'],

  vite: {
    plugins: [nxViteTsPaths()],
  },
  nitro: {
    output: {
      dir: 'dist/myapp3/.output',
      serverDir: 'dist/myapp3/.output/server',
      publicDir: 'dist/myapp3/.output/public',
    },
  },
});
"
`;

exports[`app generated files content - as-provided styles setup should configure scss 1`] = `
"import { nxViteTsPaths } from '@nx/vite/plugins/nx-tsconfig-paths.plugin';
import { defineNuxtConfig } from 'nuxt/config';

// https://nuxt.com/docs/api/configuration/nuxt-config
export default defineNuxtConfig({
  srcDir: 'src',
  buildDir: 'dist/myapp2/.nuxt',
  devtools: { enabled: true },
  devServer: {
    host: 'localhost',
    port: 4200,
  },
  typescript: {
    typeCheck: true,
    tsConfig: {
      extends: 'myapp2/tsconfig.app.json',
    },
  },
  imports: {
    autoImport: false,
  },

  css: ['~/assets/css/styles.scss'],

  vite: {
    plugins: [nxViteTsPaths()],
  },
  nitro: {
    output: {
      dir: 'dist/myapp2/.output',
      serverDir: 'dist/myapp2/.output/server',
      publicDir: 'dist/myapp2/.output/public',
    },
  },
});
"
`;

exports[`app generated files content - as-provided styles setup should not configure styles 1`] = `
"import { nxViteTsPaths } from '@nx/vite/plugins/nx-tsconfig-paths.plugin';
import { defineNuxtConfig } from 'nuxt/config';

// https://nuxt.com/docs/api/configuration/nuxt-config
export default defineNuxtConfig({
  srcDir: 'src',
  buildDir: 'dist/myapp4/.nuxt',
  devtools: { enabled: true },
  devServer: {
    host: 'localhost',
    port: 4200,
  },
  typescript: {
    typeCheck: true,
    tsConfig: {
      extends: 'myapp4/tsconfig.app.json',
    },
  },
  imports: {
    autoImport: false,
  },

  vite: {
    plugins: [nxViteTsPaths()],
  },
  nitro: {
    output: {
      dir: 'dist/myapp4/.output',
      serverDir: 'dist/myapp4/.output/server',
      publicDir: 'dist/myapp4/.output/public',
    },
  },
});
"
`;
